db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -1042,6 +1038,17 @@ public ResultSetNode bindNonVTITables(DataDictionary dataDictionary,
                                           FromList fromListParam)throws StandardException{
         TableDescriptor tableDescriptor=bindTableDescriptor();
 
+        int tableType = tableDescriptor.getTableType();
+        if(pastTxId >= 0)
+        {
+            if(tableType==TableDescriptor.VIEW_TYPE) {
```

Comment:
Can you add check for `tableDescriptor.getTableType()==TableDescriptor.WITH_TYPE` also?

Reply:
Sure, I added a check and a test as well.